tool/functiontool/function_test.go

```diff
@@ -89,15 +89,15 @@ func TestFunctionTool_Simple(t *testing.T) {
 		},
 	}
 
-	weatherReport := func(ctx tool.Context, input Args) Result {
+	weatherReport := func(ctx tool.Context, input Args) (Result, error) {
 		city := strings.ToLower(input.City)
 		if ret, ok := resultSet[city]; ok {
-			return ret
+			return ret, nil
 		}
 		return Result{
 			Status: "error",
```

Comment:
you can repurpose this test to return an error (instead of using the Status field in Result).

Reply:
Done.